src/goDocumentation.ts

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import path = require('path');
+import { CancellationToken, TextDocumentContentProvider, Uri } from 'vscode';
+import { getBinPath, getToolsEnvVars, byteOffsetAt, getFileArchive } from './util';
+
+const missingToolMsg = 'Missing tool: ';
+
+export function getDocumentation(): void {
+	let gogetdoc = getBinPath('gogetdoc');
+	if (!path.isAbsolute(gogetdoc)) {
+		vscode.window.showErrorMessage(missingToolMsg + 'gogetdoc');
+		return;
+	}
+
+	let ate = vscode.window.activeTextEditor;
+	if (typeof ate !== 'undefined') {
+		let document = ate.document;
+		let gogetDocsFlags = ['-u', '-json', '-modified', '-pos', document.fileName + ':#' + byteOffsetAt(document, ate.selection.active).toString()];
+		let p = cp.execFile(gogetdoc, gogetDocsFlags, {env: getToolsEnvVars()}, (err, stdout, stderr) => {
+			try {
+				let goGetDocOutput = <GoGetDocOutput>JSON.parse(stdout.toString());
+
+				// drop the stuff before /vendor/ if we have it
+				let imprt = goGetDocOutput.import.replace(/(.*\/vendor\/)?(.*)/, '$2');
+
+				_getDocumentationForImport(imprt);
+			} catch (e) {
+				_getDocumentationForImport('');
+			}
+		});
+		p.stdin.end(getFileArchive(document));
+	} else {
+		_getDocumentationForImport('');
+	}
+}
+
+function _getDocumentationForImport(imprt: string): void {
+	vscode.window.showInputBox({
```

Comment:
why show the inputbox at all if an import path was provided?

Reply:
This way if the cursor happens to be on some `gogetdoc`-able thing accidentally, the user has the opportunity to enter a package to view. There's one command to get docs. I suppose it might be possible to split it so that a context menu click or a keyboard shortcut invocation would trigger the "try to find the thing my cursor is on and only show me the input if you can't find anything" flow, while a more general `command-shift-p` invocation  would always show the user the box. The current "one command to rule them all" way seemed less complicated. Is there a way to detect how the command is being invoked (context menu, vs. shortcut, vs. command palette) ?

Reply:
The callback for `vscode.commands.registerCommand` takes in an array of arguments.
This would be null/undefined when the command is run from the command pallet
If the command was run from context menu, then this would have the file details corresponding to the file in the editor of the context menu.

You can also use a `quickPick` control and pre-fill it with all valid packages when no valid pkg can be inferred from gogetdoc. See https://github.com/Microsoft/vscode-go/blob/0.6.66/src/goBrowsePackage.ts#L75

Reply:
added this functionality too